ral-facilities/operationsgateway
# DSEGOG-10 sticky column headers and ID column, plus date/time selector

## Description
This PR does a lot of things :

- made column headers sticky, such that when the table is scrolled vertically, they remain in the viewport. I've simulated this by increasing the number of results per page and setting a maximum height on the table container
- made the ID column sticky, such that when the table is scrolled horizontally, the entire column and header remains in the viewport. You can simulate this by adding more columns than can fit in the viewport together
- added a date/time filter option above the table, separate from the component itself. These contain two date ranges for both start date/time and end date/time. Whenever one is either successfully filled with a valid date or cleared, the onChange method is fired, which updates the record query
- made the ID column visible at all times, with no option to remove it as per the issue details
- made the ID column non-draggable since it always needs to be on the left of the table
- recreated a lot of tests based on the above ID column changes
- added some more Cypress tests and generally tried to bump the global coverage in unit tests. _Check snapshots carefully_ as there have been many changes made and we want to make sure we get the expected results

I also noticed a bug on Friday afternoon that was present before starting this issue. When a column is added or removed, the current widths of all columns are put back to their default, erasing all user customised table sizing. I have not issued a fix for this in this PR as it's late on Friday and I wanted to get this reviewed by the time I get from leave. Either we fix it here, if it's easily doable, or we create an issue for it and attack it later.

## Testing instructions
I've tried my best to make sure everything plays together via unit tests but the amount of features in this table is quickly stacking up and will continue to lead to more unintended consequences (see above note about column resizing bug). So please test new functionality with as much existing functionality as possible.


## Agile board tracking
Closes DSEGOG-10
